### PR TITLE
[stable/stackdriver-exporter] Add support for Prometheus Operator ServiceMonitor

### PR DESCRIPTION
**Pull request by meyskens**
#### What this PR does / why we need it:
This adds an optional ServiceMonitor object to work with the Prometheus Operator

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/stackdriver-exporter/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    interval: {{ .Values.serviceMonitor.interval }}
+    {{- end }}
+    honorLabels: {{ .Values.serviceMonitor.honorLabels }}
+  selector:
+    matchLabels:
+      chart: {{ template "stackdriver-exporter.chart" . }}
+      app: {{ template "stackdriver-exporter.name" . }}
+      release: "{{ .Release.Name }}"
+      heritage: "{{ .Release.Service }}"
+{{- end -}}
```
Comment (Aisuko) [inline stable/stackdriver-exporter/templates/servicemonitor.yaml]: So cleanly work, but I believe there need to have a new line, ```{{- end }}``` would be better.

File: stable/stackdriver-exporter/values.yaml [hunk 2]
```diff
@@ -83,3 +83,12 @@ serviceAccount:
...
   name:
+
+# Enable this if you're using https://github.com/coreos/prometheus-operator
+serviceMonitor:
+  enabled: false
+  namespace: monitoring
+  # fallback to the prometheus default unless specified
+  # interval: 10s
+  ## Defaults to what's used if you follow CoreOS [Prometheus Install Instructions](https://github.com/helm/charts/tree/master/stable/prometheus-operator#tldr)
+  honorLabels: true
```
Comment (Aisuko) [inline stable/stackdriver-exporter/values.yaml:94]: Same as above.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thanks for your PR, there may have some advice, please have a look.

File: stable/stackdriver-exporter/templates/servicemonitor.yaml (hunk 1 above)
Comment (meyskens) [reply stable/stackdriver-exporter/templates/servicemonitor.yaml]: :+1:

File: stable/stackdriver-exporter/values.yaml (hunk 2 above)
Comment (meyskens) [reply stable/stackdriver-exporter/values.yaml:94]: The only other comment i could find was about `{{- end }}`, am I missing something?

**Comment by meyskens** [comment]
@Aisuko thanks for the review! Addressed the comments.

File: stable/stackdriver-exporter/values.yaml (hunk 2 above)
Comment (Aisuko) [reply stable/stackdriver-exporter/values.yaml:94]: I mean there may need a new line.

Comment (meyskens) [reply stable/stackdriver-exporter/values.yaml:94]: @Aisuko sorry think I am misreading something, there is already a newline after this line

**Comment by Aisuko** [comment]
ping @maorfr Hello sir, please help us merge this.

**Pull request merged**